Source: Nikolay Orlyuk <virkony@gmail.com>
Upstream: http://bugs.mercury.csse.unimelb.edu.au/view.php?id=259
Reason: Current Mercury sources doesn't handle libdir override well as well as
        man, info and lisp dirs. But this is crusical for multilib env.

--- mercury-srcdist-13.05/configure.ac.orig	2013-05-14 07:10:16.000000000 +0300
+++ mercury-srcdist-13.05/configure.ac	2013-05-18 12:48:16.489028416 +0300
@@ -1103,6 +1103,9 @@
 AC_SUBST(FIX_PATH_FOR_CC)
 AC_SUBST(CYGPATH)
 
+lispdir='${datarootdir}/emacs/site-lisp'
+AC_SUBST(lispdir)
+
 #-----------------------------------------------------------------------------#
 # Check for `-lm': some systems, e.g. MacOS X (Darwin), don't have it.
 # The result of this check may be overridden below.
--- mercury-srcdist-13.05/scripts/Mmake.vars.in.orig	2013-05-18 13:02:09.669064476 +0300
+++ mercury-srcdist-13.05/scripts/Mmake.vars.in	2013-05-18 13:08:41.195748087 +0300
@@ -684,24 +684,24 @@
 
 INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)@prefix@
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
-INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury
-INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
-INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PS_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
-INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/lib/mercury/html
-INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/lib/mercury/mdb
-INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/lib/mercury/elisp
+INSTALL_LIBDIR		= $(DESTDIR_AND_SLASH)@libdir@/mercury
+INSTALL_INFO_DIR	= $(DESTDIR_AND_SLASH)@infodir@
+INSTALL_DVI_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_TEXT_DIR	= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_PS_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_PDF_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_MAN_DIR		= $(DESTDIR_AND_SLASH)@mandir@
+INSTALL_HTML_DIR	= $(DESTDIR_AND_SLASH)@docdir@/html
+INSTALL_MDB_DOC_DIR	= $(DESTDIR_AND_SLASH)@docdir@/mdb
+INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)@lispdir@
 INSTALL_CGI_DIR		= $(DESTDIR_AND_SLASH)@CGIDIR@
 
 FINAL_INSTALL_PREFIX	= @prefix@
 FINAL_INSTALL_BINDIR 	= $(FINAL_INSTALL_PREFIX)/bin
-FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury
-FINAL_INSTALL_INFO_DIR	= $(FINAL_INSTALL_PREFIX)/info
-FINAL_INSTALL_MAN_DIR	= $(FINAL_INSTALL_PREFIX)/man
-FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury/elisp
+FINAL_INSTALL_LIBDIR	= @libdir@/mercury
+FINAL_INSTALL_INFO_DIR	= @infodir@
+FINAL_INSTALL_MAN_DIR	= @mandir@
+FINAL_INSTALL_ELISP_DIR	= @lispdir@
 FINAL_INSTALL_CGI_DIR	= @CGIDIR@
 
 # You should not need to override anything below here
